### PR TITLE
dotnetCorePackages.*_6_0: 6.0.0 -> 6.0.3

### PR DESCRIPTION
**Pull request by planetbeldar**
###### Description of changes

Upgrades to the latest .NET 6 release, supersedes #161285

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by corngood — APPROVED** [review]

**Comment by IvarWithoutBones** [comment]
Thanks for the work, it looks like you need to regenerate the lockfile for `ArchiSteamFarm`, `github-runner` and `ryujinx` as well.

Result of `nixpkgs-review pr 164276` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.writers</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>ArchiSteamFarm</li>
    <li>github-runner</li>
    <li>ryujinx</li>
  </ul>
</details>
<details>
  <summary>11 packages built:</summary>
  <ul>
    <li>btcpayserver</li>
    <li>discordchatexporter-cli</li>
    <li>dotnet-aspnetcore (dotnetCorePackages.aspnetcore_6_0)</li>
    <li>dotnet-runtime (dotnetCorePackages.runtime_6_0)</li>
    <li>dotnet-sdk (dotnet-sdk_6 ,dotnetCorePackages.sdk_6_0)</li>
    <li>inklecate</li>
    <li>jackett</li>
    <li>nbxplorer</li>
    <li>omnisharp-roslyn</li>
    <li>opentabletdriver</li>
    <li>pinta</li>
  </ul>
</details>

**Comment by planetbeldar** [comment]
> 3 packages failed to build:
> * ArchiSteamFarm
> * github-runner
> * ryujinx

I guess these failing packages didn't get picked up by nixpkgs-review because they're not compatible with my darwin machine?
Is there any way to know of packages that wont get built with nixpkgs-review but depend on the changes you've made?

I'll install nix on my linux machine and see what I can do about it.

**Comment by IvarWithoutBones** [comment]
> I guess these failing packages didn't get picked up by nixpkgs-review because they're not compatible with my darwin machine?

I guess so, that's pretty annoying.

> Is there any way to know of packages that wont get built with nixpkgs-review but depend on the changes you've made?

I'm not entirely sure, I have been meaning to write a script to automatically regenerate the lockfiles (at least for `buildDotnetModule` derivations), which would need that as well. Maybe @SuperSandro2000 or @Mic92 know?

In the meantime I've updated the lockfiles at [this branch](https://github.com/IvarWithoutBones/nixpkgs/commits/dotnet-sdk-update-locks), feel free to cherry-pick those commits. I've also fixed the commit order there, the `*: update dependencies` commits were made before the actual update of the sdk.

**Comment by planetbeldar** [comment]
Result of `nixpkgs-review pr 164276` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.writers</li>
  </ul>
</details>
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>ArchiSteamFarm</li>
    <li>btcpayserver</li>
    <li>discordchatexporter-cli</li>
    <li>dotnet-aspnetcore (dotnetCorePackages.aspnetcore_6_0)</li>
    <li>dotnet-runtime (dotnetCorePackages.runtime_6_0)</li>
    <li>dotnet-sdk (dotnet-sdk_6 ,dotnetCorePackages.sdk_6_0)</li>
    <li>github-runner</li>
    <li>inklecate</li>
    <li>jackett</li>
    <li>nbxplorer</li>
    <li>omnisharp-roslyn</li>
    <li>opentabletdriver</li>
    <li>pinta</li>
    <li>ryujinx</li>
  </ul>
</details>

**Review by corngood — APPROVED** [review]

**Review by IvarWithoutBones — APPROVED** [review]
Thanks for the work, I especially like what you've done with the `print-hashes` script :blush:

Tested basic functionality with a few packages, all seems to work as expected. This looks good to me :+1: 

Result of `nixpkgs-review pr 164276` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>tests.writers</li>
  </ul>
</details>
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>ArchiSteamFarm</li>
    <li>btcpayserver</li>
    <li>discordchatexporter-cli</li>
    <li>dotnet-aspnetcore (dotnetCorePackages.aspnetcore_6_0)</li>
    <li>dotnet-runtime (dotnetCorePackages.runtime_6_0)</li>
    <li>dotnet-sdk (dotnet-sdk_6 ,dotnetCorePackages.sdk_6_0)</li>
    <li>github-runner</li>
    <li>inklecate</li>
    <li>jackett</li>
    <li>nbxplorer</li>
    <li>omnisharp-roslyn</li>
    <li>opentabletdriver</li>
    <li>pinta</li>
    <li>ryujinx</li>
  </ul>
</details>

File: pkgs/development/tools/omnisharp-roslyn/default.nix [hunk 1]
```diff
@@ -62,6 +62,21 @@ let
...
+--- a/global.json
++++ b/global.json
+@@ -1,5 +1,5 @@
+ {
+     "sdk": {
+-        "version": "6.0.100"
++        "version": "6.0"
+     }
+ }
+'';
```
Comment (SuperSandro2000) [inline pkgs/development/tools/omnisharp-roslyn/default.nix:78]: Why is this not a patch file or a substituteInPlace?

Comment (planetbeldar) [reply pkgs/development/tools/omnisharp-roslyn/default.nix:78]: I didn’t know of substituteInPlace. 
I’m going to submit a PR upgrading this package soon, I’ll make use of substituteInPlace while I’m at it, thanks.

Comment (planetbeldar) [reply pkgs/development/tools/omnisharp-roslyn/default.nix:78]: #164761

**Pull request merged**